src/hdmf/build/objectmapper.py

```diff
@@ -564,6 +565,8 @@ def get_attr_value(self, **kwargs):
                 msg = ("%s '%s' does not have attribute '%s' for mapping to spec: %s"
                        % (container.__class__.__name__, container.name, attr_name, spec))
                 raise ContainerConfigurationError(msg)
+            if isinstance(attr_val, TermSetWrapper):
+                    attr_val = attr_val.item
```

Comment:
If the data is being unwrapped here in the ObjectMapper, are the changes in`H5DataIO` actually needed?

Reply:
Here was my approach to handling the write compatibility:
1. Datasets 
Write -> write_builder -> write_group and write_dataset (write_group also calls write_dataset) In write_dataset:
```
if isinstance(data, H5DataIO):
            options['io_settings'] = data.io_settings
            dataio = data
            link_data = data.link_data
            data = data.data
        if isinstance(data, TermSetWrapper):
            data = data.item
```
With DataIO we flag it after the data after the builder has been made and unwrap it. We do the same for TermSet.

2. Attributes
Attributes had to go a bit differently. If we followed the workflow of DataIo and the wrapper for the Termset, we would have to modify the attributes after the builder has been made. However, when you look at the builder the attribute has been converted to a string. So instead of <hdmf.container.TermSetWrapper object at 0x11fde7850> it was giving ‘<hdmf.container.TermSetWrapper object at 0x11fde7850>’. 

The fix I came up with was to tackle it before the builder was made via the object mapper. As you see in the lines you highlighted.